src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php

```diff
@@ -61,23 +62,42 @@ public function __construct(array $configuration, array $connectionCredentials =
             throw new LogicException('The redis transport requires php-redis 4.3.0 or higher.');
         }
 
-        $this->connection = $redis ?: new \Redis();
-        $this->connection->connect($connectionCredentials['host'] ?? '127.0.0.1', $connectionCredentials['port'] ?? 6379);
-        $this->connection->setOption(\Redis::OPT_SERIALIZER, $redisOptions['serializer'] ?? \Redis::SERIALIZER_PHP);
-
+        $host = $connectionCredentials['host'] ?? '127.0.0.1';
+        $port = $connectionCredentials['port'] ?? 6379;
+        $serializer = $redisOptions['serializer'] ?? \Redis::SERIALIZER_PHP;
+        $dbIndex = $configuration['dbindex'] ?? self::DEFAULT_OPTIONS['dbindex'];
         $auth = $connectionCredentials['auth'] ?? null;
         if ('' === $auth) {
             $auth = null;
         }
 
-        if (null !== $auth && !$this->connection->auth($auth)) {
-            throw new InvalidArgumentException('Redis connection failed: '.$this->connection->getLastError());
+        $initializer = function ($redis) use ($host, $port, $auth, $serializer, $dbIndex) {
```

Comment:
the closure should be static, to save a circular ref

Reply:
Sure. Fixed!